datadict/datadict-oci8.inc.php

```diff
@@ -185,32 +192,52 @@ function _CreateSuffix($fname,&$ftype,$fnotnull,$fdefault,$fautoinc,$fconstraint
 		return $suffix;
 	}
 
-/*
-CREATE or replace TRIGGER jaddress_insert
-before insert on jaddress
-for each row
-begin
-select seqaddress.nextval into :new.A_ID from dual;
-end;
-*/
+	/**
+	* Creates an insert trigger to emulate an auto-increment column
+	* in a table
+	*
+	* @param	str		$tabname		The name of the table
+	* @param	str[]	$tableoptions   Optional configuration items
+	*
+	* @return	str[]	The SQL statements to create the trigger
+	*/
 	function _Triggers($tabname,$tableoptions)
 	{
+		
 		if (!$this->seqField) return array();
 
-		if ($this->schema) {
+		if ($this->schema) 
+		{
 			$t = strpos($tabname,'.');
-			if ($t !== false) $tab = substr($tabname,$t+1);
-			else $tab = $tabname;
+			if ($t !== false) 
+				$tab = substr($tabname,$t+1);
+			else 
+				$tab = $tabname;
+			
+			if ($this->connection->useCompactAutoIncrements)
+				$id = sprintf('%u',crc32(strtolower($tab)));
```

Comment:
Suggest to use `hash("crc32b", $str)` instead to avoid potential issues with negative numbers (see warning in [PHP documentation](https://www.php.net/manual/en/function.crc32.php)).

However unlikely that may be, what would happen if there is a collision on the hash ?

Reply:
You don't get -ve numbers on 64 bit installs, and as far as I can see, you only get collisions on binary strings.

If I was starting from scratch  perhaps your hash idea would be better, but to be honest, I'm trying to backfill stuff from my production version into master. As you pointed out, with 12, the limitation goes away, so I would suspect it's a feature with a short lifespan.

Reply:
OK fine